reactive/kotlinx-coroutines-reactive/src/Await.kt

```diff
@@ -22,7 +23,7 @@ import kotlin.coroutines.*
  *
  * @throws NoSuchElementException if publisher does not emit any value
  */
-public suspend fun <T> Publisher<T>.awaitFirst(): T = awaitOne(Mode.FIRST)
+public suspend fun <T> Publisher<T>.awaitFirst(): T = injectCoroutineContext().awaitOne(Mode.FIRST)
```

Comment:
Just use `injectCoroutineContext` inside `awaitOne` implementation. No need to call it from every `awaitXxx` function.